src/adapters/refract/transformResource.coffee

```diff
@@ -0,0 +1,122 @@
+_ = require('./helper')
+blueprintApi = require('../../blueprint-api')
+getDescription = require('./getDescription')
+getHeaders = require('./getHeaders')
+getUriParameters = require('./getUriParameters')
+
+module.exports = (resourceElement) ->
+  resources = []
+
+  resourceDescription = getDescription(resourceElement)
+
+  _.transitions(resourceElement).forEach((transitionElement) ->
+    description = getDescription(transitionElement)
+
+    resourceParameters = getUriParameters(_.get(resourceElement, 'attributes.hrefVariables'))
+    actionParameters = getUriParameters(_.get(transitionElement, 'attributes.hrefVariables'))
+
+    attributes = _.dataStructures(resourceElement)
+    attributes = undefined if _.isEmpty(attributes)
```

Comment:
@smizell 

Agree! Just a note—I took the https://github.com/apiaryio/metamorphoses/blob/master/src/adapters/refract/transformResources.coffee#L11-L122 method and moved it to a separate file, haven't made any changes to it.